ansible_collections/arista/avd/roles/eos_designs/python_modules/l3_edge/ethernet_interfaces.py

```diff
@@ -26,20 +29,40 @@ def ethernet_interfaces(self) -> dict | None:
                 # Remove None values
                 ethernet_interface = {key: value for key, value in ethernet_interface.items() if value is not None}
 
-                interface_name = p2p_link["data"]["interface"]
-                ethernet_interfaces[interface_name] = ethernet_interface
+                if (found_eth_interface := get_item(ethernet_interfaces, "name", ethernet_interface["name"])) is None:
+                    ethernet_interfaces.append(ethernet_interface)
+                else:
+                    if found_eth_interface == ethernet_interface:
+                        # Same ethernet_interface information twice in the input data. So not duplicate interface name.
+                        continue
+
+                    raise AristaAvdError(
+                        f"Duplicate interface name {ethernet_interface['name']} found while generating ethernet_interfaces for connected_endpoints peer:"
```

Comment:
Please also fix this the remaining places, so it properly guides the user to know where to look for the duplicate.

```suggestion
                        f"Duplicate interface name {ethernet_interface['name']} found while generating ethernet_interfaces for l3_edge peer:"
```